Sources/Apodini/Semantic Model Builder/gRPC/GRPCInterfaceExporter.swift

```diff
@@ -36,105 +36,91 @@ class GRPCInterfaceExporter: InterfaceExporter {
             }
 
-            do {
-                let result: Type = try decodeParameter(from: data, with: fieldTag)
-                return result
-            } catch {
-                // Decoding fails if the parameter is not present
-                // in the payload.
-                // Not-presence of non-optional parameters is
-                // handled in shared model, so we do not care here.
-                return nil
+            app.logger.info("\tParameters:")
+            for parameter in endpoint.parameters {
+                app.logger.info("\t\t\(parameter.propertyType) \(parameter.name) = \(getFieldTag(for: parameter) ?? 0);")
             }
-        default:
-            // GRPC theoretically would also allow for other
-            // types of payload formats, e.g. JSON.
-            throw GRPCError.unsupportedContentType(
-                "Content type \(contentType ?? "") is currently not supported by Apodini GRPC exporter"
-            )
+        } catch GRPCServiceError.endpointAlreadyExists {
+            app.logger.error("Tried to overwrite endpoint \(methodName) for gRPC service \(serviceName)")
+        } catch {
+            app.logger.error("Error while exporting endpoint \(methodName) for gRPC service \(serviceName): \(error)")
         }
     }
-}
 
-// MARK: Parameter retrieval utility
+    /// The GRPC exporter handles all parameters equally as body parameters
+    func retrieveParameter<Type: Decodable>(_ parameter: EndpointParameter<Type>, for request: GRPCMessage) throws -> Type?? {
+        guard let fieldTag = getFieldTag(for: parameter) else {
+            // If this occurs, something went fundamentally wrong in usage
+            // of the GRPC exporter.
+            // Each parameter should get a default field tag assigned
+            // above in the export() funtction.
+            fatalError("No default or explicit field tag available")
+        }
 
-extension GRPCInterfaceExporter {
-    /// Returns the data contained in the body of the GRPC request.
-    private func bodyData(from request: Vapor.Request) -> Data? {
-        guard let byteBuffer = request.body.data else {
-            print("Cannot read body data.")
+        if request.data.count == 0 {
+            throw GRPCError.decodingError("""
+                No body data available to decode from.
+                GRPC exporter expects all parameters to be in the body of the request.
+                """)
+        }
+
+        do {
+            // we need to wrap the type into a struct to
+            // actually have a messsage.
+            let wrappedType = RequestWrapper<Type>.self
+            // set the fieldNumber to the one annotated at the
+            // parameter, or use default interference if none is
+            // annotated at the parameter.
+            FieldNumber.setFieldNumber(fieldTag)
+            let wrappedDecoded = try ProtoDecoder().decode(wrappedType, from: request.data)
```

Comment:
nit: when are you using `ProtoXYZ` and when `ProtobufferXYZ`?

Reply:
Thanks for pointing this out. @nityanandaz actually agreed to always use `Protobuffer` instead of abbreviations, so this has to be changed.